### PR TITLE
Mtm 36095 smartrest documentation rework

### PR DESCRIPTION
**Pull request by rssag**

### CODE REVIEW

File: content/concepts/introduction-bundle/software-libraries.md [hunk 1]
```diff
@@ -9,6 +9,6 @@ aliases:
 
 {{< product-c8y-iot >}} offers a wide range of functionality for interfacing IoT devices and other IoT-related data sources with the {{< product-c8y-iot >}} platform.
 
-We offer a generic client library for [C++](/device-sdk/cpp). If your device uses a completely proprietary runtime environment, you can always  use [MQTT](/device-sdk/mqtt) and [SmartREST 2](/device-sdk/rest). These will work on practically any Internet-connected device today, down to the smallest systems.
+We offer a generic client library for [C++](/device-sdk/cpp). If your device uses a completely proprietary runtime environment, you can always  use [MQTT](/reference/smartrest-two) and [SmartREST 2](/device-sdk/rest). These will work on practically any Internet-connected device today, down to the smallest systems.
```
Comment (BeateRixen) [inline content/concepts/introduction-bundle/software-libraries.md:12]: always  use > extra space?

File: content/device-sdk/device-mqtt-bundle/device-certificates.md [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+**Auto registration**
+
+The user for the device will be created during the first MQTT call, if a device certificate is derived from a trusted certificate which was uploaded to the {{< product-c8y-iot >}} platform with a flag _autoRegistrationEnabled_ with a value of true.
+Auto-registration needs to be activated for the uploaded certificate.
+If auto-registration is not activated it is required to use the bulk registration (see below).
+To manage the auto registration field of uploaded certificates in the UI refer to [Device Management > Managing device data > Managing trusted certificates](/users-guide/device-management#trusted-certificates).
+
+**Bulk registration**
+
+The user for the device can also be created via the standard bulk registration in Device Management.
```
Comment (BeateRixen) [inline content/device-sdk/device-mqtt-bundle/device-certificates.md:39]: Add a link here?

File: content/device-sdk/device-mqtt-bundle/device-certificates.md [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+The CSV file used in bulk registration should meet the requirements described in [Create a bulk device credentials request](https://{{< domain-c8y >}}/api/{{< c8y-current-version >}}/#operation/postBulkNewDeviceRequestCollectionResource) in the {{< openapi >}}. Moreover, it is required that the CSV file has an additional column "AUTH_TYPE" with value "CERTIFICATES", and that the column "CREDENTIALS" is either not present or has an empty value.
+
+**Single registration**
+
+Single registration is not supported for devices which are going to use certificates for authentication.
+
+>**Info:** During device registration, the device user is created, which is necessary for device communication with the platform.
+
+
+### JWT Token retrieval
```
Comment (BeateRixen) [inline content/device-sdk/device-mqtt-bundle/device-certificates.md:50]: Token > token

File: content/reference/smartrest-two-bundle/overview.md [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
 
+<!--
+based on microservice sdk smartrest.md
+TODO: make this independent from HTTP and SR1
+Explain goals
+Explain the general idea of sending CSV data that is brought in context by a template
+Explain benefits
+ -->
+
+This section walks you through the *SmartREST* protocol, the data format used, as well as the anatomy and registration of *SmartREST* templates. Built-in messages as well as errors are also discussed. For a step-by-step description, see [SmartREST](/microservice-sdk/rest#smartrest) in the *Microservice SDK guide*.
```
Comment (BeateRixen) [inline content/reference/smartrest-two-bundle/overview.md:15]: *SmartREST* > SmartREST (2x)

File: content/concepts/introduction-bundle/software-libraries.md (hunk 1 above)
Comment (rssag) [reply content/concepts/introduction-bundle/software-libraries.md:12]: done and done

File: content/device-sdk/device-mqtt-bundle/device-certificates.md (hunk 2 above)
Comment (rssag) [reply content/device-sdk/device-mqtt-bundle/device-certificates.md:39]: added

File: content/device-sdk/device-mqtt-bundle/device-certificates.md (hunk 3 above)
Comment (rssag) [reply content/device-sdk/device-mqtt-bundle/device-certificates.md:50]: done

File: content/device-sdk/device-mqtt-bundle/device-certificates.md [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+Both keystore and truststore can be stored in the same file, but for security reasons we recommended you to keep them separately.
+The truststore contains public data and the keystore contains the private key, which only to the owner should know.
+The most popular formats for these files are:
+
+* PKCS12 (Public Key Cryptography Standards, version 12), which can be generated using OpenSSL toolkit.
+* JKS (Java KeyStore), which can be generated with the Java Keytool.
+
+### Generating and signing certificates
+
+To generate certificates we will use the OpenSSL toolkit. If you do not have it already installed, then you can download if from the website: https://www.openssl.org/source/
```
[the remark this answers is not in the document]
Comment (rssag) [reply content/device-sdk/device-mqtt-bundle/device-certificates.md:140]: done

File: content/reference/smartrest-two-bundle/overview.md (hunk 4 above)
Comment (rssag) [reply content/reference/smartrest-two-bundle/overview.md:15]: done

**Review by l2p- — APPROVED** [review]

**Pull request merged**